src/components/transactions/swapComparison.js

```diff
@@ -0,0 +1,443 @@
+  ],
+};
+
+// Helper functions for input validation and parsing
+function isValidAddress(address) {
+  return ethers.utils.isAddress(address);
+}
+
+function parseEthAmount(input) {
+  const value = input.trim();
+  if (!value) {
+    return null;
+  }
+
+  try {
+    const parsed = parseFloat(value);
+    if (isNaN(parsed) || parsed < 0) {
+      return null;
+    }
+    // Convert ETH to wei and then to hex
+    return ethers.utils.parseEther(value.toString()).toHexString();
+  } catch {
+    return null;
+  }
+}
+
+function parseFeePercentage(input) {
+  const value = input.trim();
+  if (!value) {
+    return null;
+  }
+
+  try {
+    const parsed = parseFloat(value);
+    if (isNaN(parsed) || parsed < 0 || parsed > 100) {
+      return null;
+    }
+    // Convert percentage to basis points (1% = 100 basis points)
+    return Math.round(parsed * 100);
+  } catch {
+    return null;
+  }
+}
+
+function getConfigValues() {
+  const feeRecipientElement = document.getElementById('feeRecipientInput');
+  const ethAmountElement = document.getElementById('ethAmountInput');
+  const feePercentageElement = document.getElementById('feePercentageInput');
+
+  const feeRecipientInput = feeRecipientElement
+    ? feeRecipientElement.value.trim()
+    : '';
+  const ethAmountInput = ethAmountElement ? ethAmountElement.value.trim() : '';
+  const feePercentageInput = feePercentageElement
+    ? feePercentageElement.value.trim()
+    : '';
+
+  // Use defaults if inputs are empty or invalid
+  const feeRecipient =
+    feeRecipientInput && isValidAddress(feeRecipientInput)
+      ? feeRecipientInput
+      : DEFAULT_FEE_RECIPIENT;
+
+  const ethAmountHex =
+    parseEthAmount(ethAmountInput) ||
+    ethers.utils.parseEther(DEFAULT_ETH_AMOUNT).toHexString();
+
+  const feeBips =
+    parseFeePercentage(feePercentageInput) || DEFAULT_FEE_PERCENTAGE * 100;
+
+  return {
+    feeRecipient,
+    ethAmountHex,
+    feeBips,
+    ethAmountDisplay: ethAmountInput || DEFAULT_ETH_AMOUNT,
+    feePercentageDisplay:
+      feePercentageInput || DEFAULT_FEE_PERCENTAGE.toString(),
+  };
+}
+
+function createAction(action, parameters) {
+  const encodedInput = ethers.utils.defaultAbiCoder.encode(
+    V4_BASE_ACTIONS_ABI_DEFINITION[action].map((v) => v.type),
+    parameters,
+  );
+  return { action, encodedInput };
+}
+
+function addAction(type, parameters) {
+  const command = createAction(type, parameters);
+  const newParam = command.encodedInput;
+  const newAction = command.action.toString(16).padStart(2, '0');
+
+  return {
+    newParam,
+    newAction,
+  };
+}
+
+export function swapComparisonComponent(parentContainer) {
+  parentContainer.insertAdjacentHTML(
+    'beforeend',
+    `<div class="col-xl-4 col-lg-6 col-md-12 col-sm-12 col-12 d-flex align-items-stretch">
+        <div class="card full-width">
+            <div class="card-body">
+            <h4 class="card-title">
+                Swap Comparison (High Fee Test - Mainnet only)
```

Comment:
if this is intended to work on mainnet only, maybe we could simply display the component depending on which network we have selected, using `globalContext.chainIdInt`?

As at the moment, nothing prevents me to trigger the button if I'm on a different network, leading to unexpected results

<img width="1056" height="901" alt="image" src="https://github.com/user-attachments/assets/4da5c55d-07d2-4c6f-829f-4b688f0ee365" />